File: model/src/main/java/net/contrapt/auction/model/WinningBid.java
package net.contrapt.auction.model;

import javax.persistence.*;
import java.math.BigDecimal;

/**
 * Created by msimmons on 10/10/14.
 */
@Entity
@Table(name = "winning_bid")
public class WinningBid extends AbstractEntity {

    //@ManyToOne(fetch = FetchType.EAGER, optional = false)
    //private Bidder bidder;
    private Long bidderId;

    //@ManyToOne(fetch = FetchType.EAGER, optional = false)
    //private Item item;
    private Long itemId;

    @Column
    private BigDecimal amount;

    protected WinningBid() {}

    public WinningBid(Item item, Bidder bidder, BigDecimal amount) {
        this.itemId = item.getId();
        this.bidderId = bidder.getId();
        this.amount = amount;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public Long getItemId() {
        return itemId;
    }

    public Long getBidderId() {
        return bidderId;
    }

    @Override
    public String uniqueKey() {
        return bidderId+":"+itemId;
    }
}

File: model/src/main/kotlin/net/contrapt/auction/model/WinningBid.kt
package net.contrapt.auction.model

import java.math.BigDecimal
import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

/**
 * Created by msimmons on 10/10/14.
 */
@Entity
@Table(name = "winning_bid")
class WinningBid(item : Item, bidder : Bidder, val amount : BigDecimal) : AbstractEntity() {

    @ManyToOne
    @JoinColumn(name = "itemId")
    val item : Item

    @ManyToOne
    @JoinColumn(name= "bidderId")
    val bidder : Bidder

    init {
        this.item = item
        this.bidder = bidder
    }

    protected constructor() : this(Item(), Bidder(), BigDecimal.ZERO){}

    override fun uniqueKey(): String {
        return "$bidder.id:$item.id"
    }
}
